ci_framework/roles/cifmw_cephadm/tasks/bootstrap.yml

```diff
@@ -98,6 +98,18 @@
         {{ cifmw_cephadm_container_ns + '/' + cifmw_cephadm_container_image + ':' + cifmw_cephadm_container_tag|string }}
       changed_when: false
 
+    - name: Switch HA container images if a custom container registry is used
+      become: true
+      when:
+        - cifmw_cephadm_registry_url | length > 0
+        - cifmw_cephadm_container_ns != "quay.io/ceph"
+      ansible.builtin.shell: |
+        {{ cifmw_cephadm_ceph_cli }} config set mgr mgr \
+          mgr/cephadm/container_image_keepalived {{ cifmw_cephadm_container_ns + '/keepalived:latest' }}
+        {{ cifmw_cephadm_ceph_cli }} config set mgr mgr \
+          mgr/cephadm/container_image_haproxy {{ cifmw_cephadm_container_ns + '/haproxy:latest' }}
+        {{ cifmw_cephadm_ceph_cli }} orch redeploy ingress.rgw.default
```

Comment:
We should not need redeploy. Maybe you had it because you wanted to update it after it was deployed, but once you get a fresh deployment, the redeploy won't be needed. Would you please delete line 111?

Reply:
Right, basically we have three phases during the ceph deployment:
1. bootstrap
2. apply spec
3. deploy additional daemons, pools, keys, etc

During the second phase, [ceph_config_set](https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/cifmw_cephadm/tasks/apply_spec.yml#L48) is called before applying the spec via the orchestrator, and if we specify the container images (`ceph config set mgr mgr mgr/cephadm/container_image_{haproxy, keepalived} ....`) using that set of tasks, when RGW (phase 3) is deployed, we should be able to get the right container images/tags.